kairos-cli/bin/crypto/signer.rs

```diff
@@ -0,0 +1,54 @@
+use std::path::PathBuf;
+
+use super::private_key::{parse_private_key, CasperPrivateKey};
+use super::public_key::CasperPublicKey;
+use crate::crypto::error::CryptoError;
+use casper_types::{bytesrepr::ToBytes, SecretKey};
+use casper_types::{crypto, PublicKey};
+
+pub struct CasperSigner {
+    secret_key: CasperPrivateKey,
+    public_key: CasperPublicKey,
+}
+
+#[allow(unused)]
+impl CasperSigner {
+    fn from_key_raw(secret_key: CasperPrivateKey) -> Self {
+        // Derive the public key.
+        let public_key = CasperPublicKey::from_key(PublicKey::from(&secret_key.0));
+
+        CasperSigner {
+            secret_key,
+            public_key,
+        }
+    }
+
+    pub fn from_file(secret_key_path: &str) -> Result<Self, CryptoError> {
```

Comment:
This and `from_key_pathbuf` are duplicates.
Call `crate::crypto::private_key::CasperPrivateKey::from_file`  instead of calling `casper_types::from_file`.
If we have our own `CasperPrivateKey` newtype we should use it consistently.

Reply:
Fixed in https://github.com/cspr-rad/kairos/pull/17/commits/b1a78b47dfce7f834e6d5f05e824b217beffa53f.